### ISSUE TITLE
In the preview link have to be only the "slug" field and not all the path

### ISSUE THREAD

**Issue by rodrigobarona** [wbobeirne/datocms-plugin-nextjs-preview#2]
Hi

I tried the plugin and I love the ideia. Congratulations.

But if I put only "$slug" works perfect on the Preview link but not on the Publish link.

If I put in the field "/post/$slug" works great on the Publish link but not on the preview because "slug=/posts/post-1" is an "Invalid slug"

Can you make a substring on the preview link?
https://github.com/wbobeirne/datocms-plugin-nextjs-preview/blob/master/src/Main.jsx#L98

Something like this: https://stackoverflow.com/a/10568818

I design in the field "/post/$slug" but in the path for the preview you make a substring all before "$".

Works?

Thanks

**Comment by wbobeirne** [comment]
Hey @rodrigobarona, I've just reviewed the DatoCMS example code and I see what's going on here. I didn't realize that in their example, they hard-code the path for previewing posts, and make sure it's a real post before showing it to you.

The way I've setup my Next.js sites is to just redirect to the path provided in the `slug` query param after setting the preview mode, since my sites have multiple content types you would want to preview, not just a single post type that we can lookup.

I'd definitely like to make this work for the demo example, but I don't want to break the use case I have.

**Comment by rodrigobarona** [comment]
Hi @wbobeirne thanks for your feedback. I see what you are talking about.

Looks like the problem is in the hard-code post link
  res.writeHead(307, { Location: `/posts/${post.slug}` })
https://github.com/vercel/next.js/blob/4ba3607dcbb6ad88a9ee6935504f2dadc91dd0de/examples/cms-datocms/pages/api/preview.js#L26

It's correct?

Thanks

**Comment by wbobeirne** [comment]
Yep, so basically in my code I have removed the check to see if it's a real post:
https://github.com/vercel/next.js/blob/4ba3607dcbb6ad88a9ee6935504f2dadc91dd0de/examples/cms-datocms/pages/api/preview.js#L13-L19

And I have changed the line you linked to
```ts
res.writeHead(307, { Location: post.slug })
```
However I realize now that that makes very little sense from a nomenclature standpoint, since `slug` refers to the value, what I'm _really_ passing here is a `path`.

It would be a breaking change, but I should probably change it to that, and update the docs to inform people of this behavior. Or let them choose the name of the parameter. But I can't just send `$slug`, because what if someone has a path that has multiple variables, like `/posts/$date/$slug` or `/$language_code/posts/$slug` etc.

Perhaps the other alternative would be to send over the whole path, and then individual fields, like
```js
req.query = {
  path: "/posts/02-25-20/my-cool-post",
  slug: "my-cool-post",
  date: "02-25-20"
}
```
That way you could do whichever technique you wanted.

**Comment by rodrigobarona** [comment]
Hi @wbobeirne 

Thanks for the explanation, and I agree with your point, because I want to build a multilingual website and I want to see the preview Page in ENG and SPA for example.

For that reason, reimagine the options to pass the whole path make sense.

Let me know if I can help!

Thanks
